src/form/lpac.jl

```diff
@@ -130,3 +130,254 @@ function constraint_ohms_yt_to(pm::AbstractLPACCModel, n::Int, f_bus, t_bus, f_i
     JuMP.@constraint(pm.model, q_to == -(b+b_to)*(1.0 + 2*phi_to) - (-b*tr+g*ti)/tm^2*(cs + phi_fr + phi_to) + (-g*tr-b*ti)/tm^2*-(va_fr-va_to) )
 end
 
+"`angmin*branch_ne[i] + vad_min*(1-branch_ne[i]) <= t[f_bus] - t[t_bus] <= angmax*branch_ne[i] + vad_max*(1-branch_ne[i])`"
+function constraint_ne_voltage_angle_difference(pm::AbstractLPACCModel, n::Int, f_idx, angmin, angmax, vad_min, vad_max)
+    i, f_bus, t_bus = f_idx
+
+    va_fr = var(pm, n, :va, f_bus)
+    va_to = var(pm, n, :va, t_bus)
+    z = var(pm, n, :branch_ne, i)
+    JuMP.@constraint(pm.model, va_fr - va_to <= angmax*z + vad_max*(1-z))
+    JuMP.@constraint(pm.model, va_fr - va_to >= angmin*z + vad_min*(1-z))
+end
+
+""
+function variable_ne_branch_voltage(pm::AbstractLPACCModel; kwargs...)
+    variable_ne_branch_voltage_magnitude_fr(pm; kwargs...)
+    variable_ne_branch_voltage_magnitude_to(pm; kwargs...)
+    variable_ne_branch_voltage_angle_fr(pm; kwargs...)
+    variable_ne_branch_voltage_angle_to(pm; kwargs...)
```

Comment:
These constraints should reason over voltage angle differences `vad` and not specific angles, `va_fr` and `va_to`. 

See `variable_branch_voltage_product_angle_on_off` as an example, which you can likely reuse.

Reply:
Thanks for the review @ccoffrin. I have a better idea of what needs to be done now. I would try to resolve them in a week or so.